src/plugins/core/cell.ts

```diff
@@ -571,16 +571,19 @@ export class CellPlugin extends CorePlugin<CoreState> implements CoreState {
     style: Style | undefined
   ): LiteralCell {
     const locale = this.getters.getLocale();
+    const parsedValue = parseLiteral(content, locale);
+
     format = format || detectDateFormat(content, locale) || detectNumberFormat(content);
     if (format !== PLAIN_TEXT_FORMAT && !isEvaluationError(content)) {
-      content = toString(parseLiteral(content, locale));
+      content = toString(parsedValue);
     }
     return {
       id,
       content,
       style,
       format,
       isFormula: false,
+      parsedValue,
```

Comment:
we could maybe even store the full `EvaluatedCell` object. It would avoid recreating it every time (and garbage collecting the previous one).
WDYT?

Reply:
Not a big fan of putting an evaluation related object back to the core plugins ...

Reply:
Two steps further would be to do the same thing for formulas without references ? 🤔

Reply:
> ld be to do the same thing for formulas without r

doesn't work because a formula without references can still perform RPCs and depends on non-core stuff

Reply:
Doesn't cost too much to see the impact on a real-world use case ;) I also am curious on the imact it'd have on rng's spreadsheet

Reply:
### Test with `makeLargeDataset(26, 10_000, ["numbers"])`

|                            | Master | Changes on is Number | Changes on is number + store evaluated cell when literal cell |
|----------------------------|--------|----------------------|--------------------------------------------------------------|
| **Evaluate all cells**     | 254    | 148                  | 107                                                          |
| **Model created in**       | 1263   | 1045                 | 1114                                                         |

### Test on real usecase (the monstrous RNG spreadsheet)

|                            | Master | Changes on is Number | Changes on is number + store evaluated cell when literal cell |
|----------------------------|--------|----------------------|--------------------------------------------------------------|
| **Evaluate all cells**     | 743    | 719                  | 703                                                          |
| **Model created in**       | 4071   | 3792                 | 3664                                                         |


*All measurements are averages of 5 values.*